iree/turbine/kernel/wave/promotion.py

```diff
@@ -122,6 +122,13 @@ def promote_node(
     symbolic_shape = node.type.symbolic_shape
     with node.graph.inserting_after(node.graph._root):
         constrained_shape = get_constrained_shape(symbolic_shape, constraints)
+        # If the read/write operation already has a set distributed shape at the kernel
+        # we use that for allocation. Otherwise deduce the shape from constraints.
+        memory_node = get_custom(node.memory)
+        if isinstance(memory_node, Allocate) and hasattr(
```

Comment:
doesnt allocate always have the distributed_shape attribute? Shouldnt you instead check if it is None?